syncer/state_syncer.go

```diff
@@ -35,7 +35,8 @@ func minLayer(a, b types.LayerID) types.LayerID {
 
 func (s *Syncer) stateSynced() bool {
 	current := s.ticker.CurrentLayer()
-	return current.Uint32() <= 1 || !s.mesh.ProcessedLayer().Before(current.Sub(1))
+	return current <= types.GetEffectiveGenesis() ||
+		(s.mesh.ProcessedLayer() >= current-1 && !s.stateErr.Load())
```

Comment:
is there a reason why checkMeshAgreement is based on this function instead of `synced` state?

Reply:
changed.